src/decoder/stream.rs

```diff
@@ -467,6 +471,20 @@ impl StreamingDecoder {
         self.decode_config.set_ignore_text_chunk(ignore_text_chunk);
     }
 
+    /// Return whether the decoder is set to ignore the Adler-32 checksum.
+    pub fn ignore_adler32_checksum(&self) -> bool {
+        self.inflater.ignore_adler32()
+    }
+
+    /// Set whether to compute and verify the Adler-32 checksum during
+    /// decompression. The decoder defaults to `true`.
+    ///
+    /// This flag cannot be modified after decompression has started until the
+    /// [StreamingDecoder] is reset.
+    pub fn set_ignore_adler32_checksum(&mut self, flag: bool) {
+        self.inflater.set_ignore_adler32(flag);
+    }
```

Comment:
Regarding API, that's fine but there should be similar setters for `Decoder`, in its `decoder: DecodeConfig` attribute. In a more forward looking note, in `gif` we switched the structus so that the reader is supplied after the `Decoder` is configured. (The struct has been renamed to [`DecodeOptions`](https://docs.rs/gif/latest/gif/struct.DecodeOptions.html) afterwards). In that case, the api could be 

* `StreamingDecoder::set_options(&mut self, _: &DecodeOptions)`.

We should probably switch to a similar API in the next major version.

Reply:
I noticed that struct but it's `pub(crate)`, should that be made public? And should I rename it to match the `gif` API?